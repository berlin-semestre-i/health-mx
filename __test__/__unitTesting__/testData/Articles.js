const Articles = {
  articles: [
    {
      title: 'News Article 1',
      url: 'https://news.article.com/1',
      urlToImage: 'https://news.article.com/images/1',
      __typename: 'Article',
    },
    {
      title: 'News Article 2',
      url: 'https://news.article.com/2',
      urlToImage: 'https://news.article.com/images/2',
      __typename: 'Article',
    },
    {
      title: 'News Article 3',
      url: 'https://news.article.com/3',
      urlToImage: 'https://news.article.com/images/3',
      __typename: 'Article',
    },
    {
      title: 'News Article 4',
      url: 'https://news.article.com/4',
      urlToImage: 'https://news.article.com/images/4',
      __typename: 'Article',
    },
  ],
}

export default Articles