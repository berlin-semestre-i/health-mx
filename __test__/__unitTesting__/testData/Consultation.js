const newMedication = {
  dose: '20ml',
  medicine: 'Dramamine',
  frequency: '8 hrs',
  duration: '4 días',
}
const newStudy = {
  study: 'Rayos X',
  indications: 'Sin indicaciones',
}
const errorMessage = 'Llena todos los campos.'
const somatometry = {
  nurse: 'Rodrigo Fernández',
  height: 169,
  weight: 63,
  pulse: 68,
  imc: 22.06,
  temperature: 36,
}

export { newMedication, newStudy, errorMessage, somatometry }