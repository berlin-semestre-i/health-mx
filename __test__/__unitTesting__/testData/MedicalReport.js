const beneficiary = {
  firstName: 'María del Carmen',
  fatherLastName: 'Hinojosa',
  motherLastName: 'Ramírez',
  nss: '8454321878',
  curp: 'JP0A170297SLMRH00',
  clinic: 'Zapopan 36',
  umf: 'Zapopan UMF2',
  dependents: 0,
  allergies: ['Penicilina'],
  ailment: [],
  ahd: ['Diabetes tipo 2 (padre)', 'Cáncer de colon (abuelo)'],
  doctor: 'César Rengiffo Pérez',
}
const studies = [{
  date: '12 Jul, 2018',
  area: 'Radiología',
  type: 'Rayos X',
  indications: 'Bla bla bla',
}]
const treatments = [{
  consultation: '03 Nov, 2017',
  startDate: '03 Nov, 2017',
  endDate: '11 Nov 2017',
  medication: ['Paracetamol', 'Diclofenaco sódico', 'Celebrex'],
}]
const consultations = [{
  date: '12 Jul, 2018',
  area: 'Medicina familiar',
  doctor: 'Luis Alfredo Ramírez',
  status: 'Regular',
}]
const tabsValues = [
  {name: 'prev-consultations', items: consultations},
  {name: 'lab-studies', items: studies},
  {name: 'treatments', items: treatments},
]

export { beneficiary, studies, treatments, consultations, tabsValues }