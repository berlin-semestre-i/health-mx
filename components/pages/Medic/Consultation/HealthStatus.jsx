import React from 'react'
import { Dropdown } from 'semantic-ui-react'
import Card from '../../../shared/Card'

const healthStatusOptions = [{
  text: 'Regular', value: 0,
}, {
  text: 'CrÃ­tico', value: 1,
}]

const HealthStatus = ({ healthStatus, onChange }) => (
  <React.Fragment>
    <Card header="Estado de Salud">
      <Dropdown
        name="healthStatus"
        value={healthStatus}
        placeholder="Seleccionar Estado"
        fluid selection
        options={healthStatusOptions}
        onChange={onChange}
      />
    </Card>
  </React.Fragment>
)

export default HealthStatus