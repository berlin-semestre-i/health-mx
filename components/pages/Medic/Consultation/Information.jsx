import React from 'react'
import { Form } from 'semantic-ui-react'
import Card from '../../../shared/Card'
import styled from 'styled-components'

const Information = ({ motive, physicalExploration, conclusion, onChange }) => (
  <React.Fragment>
    <Card header="Información de Consulta" className="nth-col-first">
      <Form name="information">
        <FormTextArea
          name="motive"
          label="Motivo de la consulta:"
          placeholder="Escribe aquí el motivo de la consulta."
          value={motive}
          onChange={onChange}
        />
        <FormTextArea
          name="physicalExploration"
          label="Exploración física:"
          placeholder="En caso de aplicar,
                    ingresa las observaciones de la exploración física."
          value={physicalExploration}
          onChange={onChange}
        />
        <FormTextArea
          name="conclusion"
          label="Observaciones y conclusiones de la consulta:"
          placeholder="Escribe aquí las observaciones finales de la consulta."
          value={conclusion}
          onChange={onChange}
        />
      </Form>
    </Card>
  </React.Fragment>
)

export default Information

const FormTextArea = styled(Form.TextArea)`
   & textarea {
    font-family: -apple-system,system-ui,BlinkMacSystemFont, 
    "Segoe UI",Roboto,'Helvetica Neue',Arial,Helvetica,sans-serif;
   }
`