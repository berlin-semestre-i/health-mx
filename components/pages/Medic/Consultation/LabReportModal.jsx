import React, { PureComponent } from 'react'
import styled from 'styled-components'
import media from 'styled-media-query'
import { Modal, Form, Icon, Table, Button } from 'semantic-ui-react'
import Link from 'next/link'

const priorityOptions = [{
  text: 'Baja', value: 0,
}, {
  text: 'Regular', value: 1,
}, {
  text: 'Alta', value: 2,
}]

const errorMsg = 'Llena todos los campos.'

class LabReportModal extends PureComponent {

  state = {
    studies: [],
    study: '',
    indications: '',
    error: '',
    isMouseOnRow: -1,
    observations: '',
    priority: '',
  }

  updateState = (e, {name, value}) => this.setState({ [name]: value })

  mouseEnter = index => this.setState({ isMouseOnRow: index })

  mouseLeave = () => this.setState({ isMouseOnRow: -1 })

  deleteStudy = index => {
    let arrayCopy = [...this.state.studies]
    arrayCopy.splice(index, 1)
    this.setState({ studies: arrayCopy })
  }

  addStudy = () => {
    const { study, indications } = this.state

    let enabled =
      study.length > 0 &&
      indications.length > 0

    if(enabled) {
      this.setState(prevState => ({
        studies: [...prevState.studies, { study, indications }],
        study: '',
        indications: '',
        error: '',
      }))
    } else {
      this.setState({ error: errorMsg })
    }
  }

  saveStudies = () => {
    const { studies, priority, observations } = this.state
    this.props.save(studies, priority, observations)
  }

  close = () => {
    this.setState({
      study: '',
      indications: '',
      error: '',
    })
    this.props.close()
  }

  render () {

    const { study, indications, error, isMouseOnRow, studies,
      observations, priority } = this.state

    const { open } = this.props

    return (
      <React.Fragment>
        <StudiesModal open={open} onClose={this.close}>
          <ModalHeader>
            <h1 className="ui header">Nueva solicitud de estudios</h1>
            <CloseModal name="times" onClick={this.close} />
          </ModalHeader>
          <Modal.Content>
            <Form>
              <Form.Group name="newStudy">
                <Form.Input
                  name="study"
                  label="Estudio:"
                  placeholder="Escribe y selecciona el estudio"
                  width={6}
                  value={study}
                  onChange={this.updateState}
                />
                <Form.Input
                  name="indications"
                  label="Indicaciones del estudio:"
                  placeholder="Escribe las indicaciones para el estudio"
                  width={9}
                  value={indications}
                  onChange={this.updateState}
                />
                <AddIconContainer>
                  <AddIcon
                    name="plus"
                    onClick={() => this.addStudy()}
                  />
                </AddIconContainer>
              </Form.Group>
              <ErrorMessage name="error">{ error }</ErrorMessage>
              <Table basic="very">
                <Table.Body>
                  {studies.map((report, index) => (
                    <Table.Row
                      key={index}
                      onMouseEnter={() => this.mouseEnter(index)}
                      onMouseLeave={() => this.mouseLeave()}
                    >
                      <Table.Cell width={6}>{report.study}</Table.Cell>
                      <Table.Cell width={9}>
                        {report.indications.length > 0?
                          report.indications : 'Sin indicaciones'
                        }
                      </Table.Cell>
                      <Table.Cell width={1}>
                        {isMouseOnRow === index ?
                          <DeleteIcon
                            name="trash"
                            onClick={() => this.deleteStudy(index)}
                          /> : null
                        }
                      </Table.Cell>
                    </Table.Row>
                  ))}
                </Table.Body>
              </Table>
              <Form.Dropdown
                name="priority"
                value={priority}
                label="Prioridad"
                placeholder="Seleccionar Prioridad"
                fluid selection
                options={priorityOptions}
                width={6}
                onChange={this.updateState}
              />
              <Form.TextArea
                name="observations"
                label="Observaciones generales:"
                placeholder="Observaciones para realizaciÃ³n de estudio"
                value={observations}
                onChange={this.updateState}
              />
            </Form>
          </Modal.Content>
          <ModalFooter>
            <Link href="">
              <Print href="">Imprimir orden</Print>
            </Link>
            <Button basic color="blue" content="Cancelar" onClick={this.close} />
            <Button name="save-study" content="Guardar" onClick={this.saveStudies} />
          </ModalFooter>
        </StudiesModal>
      </React.Fragment>
    )
  }
}

export default LabReportModal

const ModalHeader = styled(Modal.Header)`
  &&.header {
    border-bottom: none;
    position: relative;
  }
`
const ModalFooter = styled(Modal.Actions)`
  &&.actions {
    border-top: none;
    background-color: #FFFF;
  }
`
const StudiesModal = styled(Modal)`
  & {
    padding: 1em;
  }
`
const Print = styled.a`
  float: left;
`
const CloseModal = styled(Icon)`
  & {
    position: absolute;
    top: 1em;
    right: 1em;
    cursor: pointer;
    color: #3d5170;
  }
`
const AddIconContainer = styled(Form.Field)`
position: relative;

${media.lessThan('small')`
  &&&.field {
    margin-bottom: 1em;
  }
`}
`
const AddIcon = styled(Icon)`
  && {
    color: #3d5170;
    cursor: pointer;
  }

  &&:hover {
    color: #818ea3;
  }

  ${media.greaterThan('small')`
    &&, &&:hover {
      position: absolute;
      bottom: 15px;
    }
  `}
`
const DeleteIcon = styled(Icon)`
    && {
      color: #3d5170;
      cursor: pointer;
    }

    &&:hover {
      color: #818ea3;
    }
`
const ErrorMessage = styled.span`
  color: red;
`