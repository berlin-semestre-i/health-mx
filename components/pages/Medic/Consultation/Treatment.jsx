import React, { PureComponent } from 'react'
import { Form, Table, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import media from 'styled-media-query'
import Card from '../../../shared/Card'

const errorMsg = 'Llena todos los campos.'

class Treatment extends PureComponent {

  state = {
    dose: '',
    medicine: '',
    frequency: '',
    duration: '',
    error: '',
    isMouseOnRow: -1,
  }

  updateState = (e, {name, value}) => this.setState({ [name]: value })

  deleteMedication = index => this.props.deleteMedication(index)

  addNewMedication = () => {
    const { dose, medicine, frequency, duration } = this.state

    let enabled =
      dose.length > 0 &&
      medicine.length > 0 &&
      frequency.length > 0 &&
      duration.length > 0

    if(enabled) {
      this.props.addMedication({ dose, medicine, frequency, duration })
      this.setState({
        dose: '',
        medicine: '',
        frequency: '',
        duration: '',
        error: '',
      })
    } else {
      this.setState({ error: errorMsg })
    }
  }

  mouseEnter = index => this.setState({ isMouseOnRow: index })

  mouseLeave = () => this.setState({ isMouseOnRow: -1 })

  render () {

    const { dose, medicine, frequency, duration, error,
      indications, isMouseOnRow } = this.state

    const { treatment, onChange } = this.props

    return (
      <React.Fragment>
        <Card header="Tratamiento">
          <Form>
            <Form.Group name="newMedication">
              <Form.Input
                name="dose"
                label="Dosis"
                placeholder="Dosis"
                width={2}
                value={dose}
                onChange={this.updateState}
              />
              <Form.Input
                name="medicine"
                label="Medicamento"
                placeholder="Medicamento"
                width={6}
                value={medicine}
                onChange={this.updateState}
              />
              <Form.Input
                name="frequency"
                label="Frecuencia"
                placeholder="Frecuencia"
                width={4}
                value={frequency}
                onChange={this.updateState}
              />
              <Form.Input
                name="duration"
                label="Duración"
                placeholder="Duración"
                width={3}
                value={duration}
                onChange={this.updateState}
              />
              <AddIconContainer>
                <AddIcon
                  name="plus"
                  onClick={() => this.addNewMedication()}
                />
              </AddIconContainer>
            </Form.Group>
            <ErrorMessage name="error">{ error }</ErrorMessage>
            <Table basic="very">
              <Table.Body>
                {treatment.map((med, index) => (
                  <Table.Row
                    name="medication"
                    key={index}
                    onMouseEnter={() => this.mouseEnter(index)}
                    onMouseLeave={() => this.mouseLeave()}
                  >
                    <Table.Cell width={2}>{med.dose}</Table.Cell>
                    <Table.Cell width={6}>{med.medicine}</Table.Cell>
                    <Table.Cell width={4}>{med.frequency}</Table.Cell>
                    <Table.Cell width={3}>{med.duration}</Table.Cell>
                    <Table.Cell width={1}>
                      {isMouseOnRow === index ?
                        <DeleteIcon
                          name="trash"
                          onClick={() => this.deleteMedication(index)}
                        /> : null
                      }
                    </Table.Cell>
                  </Table.Row>
                ))}
              </Table.Body>
            </Table>
            <FormTextArea
              name="indications"
              label="Indicaciones del tratamiento:"
              placeholder="Escribe aquí las indicaciones del tratamiento."
              value={indications}
              onChange={onChange}
            />
          </Form>
        </Card>
      </React.Fragment>
    )
  }
}

export default Treatment

const AddIcon = styled(Icon)`
  && {
    color: #3d5170;
    cursor: pointer;
  }

  &&:hover {
    color: #818ea3;
  }

  ${media.greaterThan('small')`
    &&, &&:hover {
      position: absolute;
      bottom: 15px;
    }
  `}
`
const DeleteIcon = styled(Icon)`
    && {
      color: #3d5170;
      cursor: pointer;
    }

    &&:hover {
      color: #818ea3;
    }
`
const AddIconContainer = styled(Form.Field)`
  position: relative;

  ${media.lessThan('small')`
    &&&.field {
      margin-bottom: 1em;
    }
  `}
`
const ErrorMessage = styled.span`
  color: red;
`
const FormTextArea = styled(Form.TextArea)`
   & textarea {
    font-family: -apple-system,system-ui,BlinkMacSystemFont, 
    "Segoe UI",Roboto,'Helvetica Neue',Arial,Helvetica,sans-serif;
   }
`