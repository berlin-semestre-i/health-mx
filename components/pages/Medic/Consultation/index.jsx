import React, { PureComponent } from 'react'
import { Grid, Button, List } from 'semantic-ui-react'
import Header from '../../../shared/PageHeader'
import Card from '../../../shared/Card'
import styled from 'styled-components'
import TreatmentCard from './Treatment'
import InformationCard from './Information'
import SomatometryCard from './Somatometry'
import HealthStatusCard from './HealthStatus'
import LabReportsModal from './LabReportModal'
import Link from 'next/link'

const somatometry = {
  nurse: 'Rodrigo Fernández',
  height: 169,
  weight: 63,
  pulse: 68,
  imc: 22.06,
  temperature: 36,
}

class Consultation extends PureComponent {

  state = {
    healthStatus: 0,
    indications: '',
    treatment: [],
    labReports: [],
    labReportsPriority: '',
    labReportsObservations: '',
    open: false,
    motive: '',
    physicalExploration: '',
    conclusion: '',
  }

  open = () => this.setState({ open: true })

  close = () => this.setState({ open: false })

  updateState = (e, {name, value}) => this.setState({ [name]: value })

  addMedication = newMedication => {
    this.setState(prevState => ({
      treatment: [...prevState.treatment, newMedication],
    }))
  }

  deleteMedication = index => {
    let arrayCopy = [...this.state.treatment]
    arrayCopy.splice(index, 1)
    this.setState({ treatment: arrayCopy })
  }

  saveLabReports = (reports, priority, observations) => {
    this.setState({
      labReports: reports,
      labReportsPriority: priority,
      labReportsObservations: observations,
      open: false,
    })
  }

  render() {
    const { healthStatus, motive, physicalExploration,
      conclusion, labReports, open, treatment } = this.state
    const { date } = this.props

    return (
      <React.Fragment>
        <Grid>
          <Header
            title="Atrás"
            subtitle={`Consulta - ${date}`}
            goBack="true" address="/medic/patient-profile"
          />
          <Grid.Row>
            <Grid.Column mobile={16} computer={5}>
              <SomatometryCard somatometry={somatometry} />
              <HealthStatusCard healthStatus={healthStatus} onChange={this.updateState} />
              <Card
                className="lab-studies"
                header="Estudios Clínicos"
                iconName="pencil"
                iconColor="blue"
                callbackFn={this.open}
              >
                {labReports.length > 0 ? (
                  <List>
                    {labReports.map((report, index) => (
                      <List.Item key={index}>
                        <List.Icon name="eye" />
                        <List.Content>{report.study}</List.Content>
                      </List.Item>
                    ))}
                  </List>
                ) : (
                  <p>No se ha añadido ninguna solicitud.</p>
                )}
              </Card>
            </Grid.Column>
            <Grid.Column mobile={16} computer={11}>
              <InformationCard
                onChange={this.updateState}
                motive={motive}
                physicalExploration={physicalExploration}
                conclusion={conclusion}
              />
              <TreatmentCard
                onChange={this.updateState}
                addMedication={this.addMedication}
                deleteMedication={this.deleteMedication}
                treatment={treatment}
              />
            </Grid.Column>
          </Grid.Row>
          <Grid.Row>
            <FinalizeButtonContiainer>
              <Link href="/medic/profile">
                <Button content="Finalizar Consulta" />
              </Link>
            </FinalizeButtonContiainer>
          </Grid.Row>
        </Grid>
        <LabReportsModal
          open={open}
          save={this.saveLabReports}
          close={this.close}
        />
      </React.Fragment>
    )
  }
}

export default Consultation

const FinalizeButtonContiainer = styled(Grid.Column)`
  text-align: right;
`