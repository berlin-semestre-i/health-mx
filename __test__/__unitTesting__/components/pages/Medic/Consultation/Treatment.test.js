import React from 'react'
import { shallow, mount } from 'enzyme'
import Treatment from '../../../../../../components/pages/Medic/Consultation/Treatment'
import { newMedication, errorMessage } from '../../../../testData/Consultation'

describe('Treatment', () => {
  it('renders', () => {
    const component = shallow(<Treatment treatment={[]} />)
    expect(component).toMatchSnapshot()
  })

  it('toggle delete button', () => {
    const add = jest.fn()
    const wrapper = mount(<Treatment
      treatment={[newMedication]}
      addMedication={add} />)
    const tableRow = wrapper.find('tr')
    tableRow.simulate('mouseenter')
    expect(wrapper.state('isMouseOnRow')).toBe(0)
    tableRow.simulate('mouseleave')
    expect(wrapper.state('isMouseOnRow')).toBe(-1)
    wrapper.unmount()
  })

  it('adds medication', () => {
    const add = jest.fn()
    const wrapper = mount(<Treatment
      treatment={[]}
      addMedication={add} />)
    wrapper.setState(newMedication)
    wrapper.find('i.plus').simulate('click')
    expect(add).toHaveBeenCalled
  })

  it('updates state', () => {
    const updateState =  jest.fn()
    const wrapper = mount(<Treatment treatment={[]} />)
    wrapper.find('input[name="dose"]').simulate('change')
    expect(updateState).toHaveBeenCalled
  })

  it('deletes medication', () => {
    const deleteMed = jest.fn()
    const wrapper = mount(<Treatment
      treatment={[newMedication]}
      deleteMedication={deleteMed} />)
    wrapper.setState({isMouseOnRow: 0})
    wrapper.find('i.trash').simulate('click')
    expect(deleteMed).toHaveBeenCalled
  })

  it('shows error message', () => {
    const wrapper = mount(<Treatment treatment={[]} />)
    wrapper.find('i.plus').simulate('click')
    expect(wrapper.state('error')).toBe(errorMessage)
  })
})