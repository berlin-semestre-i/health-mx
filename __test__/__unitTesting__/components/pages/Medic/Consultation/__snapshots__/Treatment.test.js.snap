// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Treatment renders 1`] = `
<Fragment>
  <CustomCard
    header="Tratamiento"
  >
    <Form
      as="form"
    >
      <FormGroup
        name="newMedication"
      >
        <FormInput
          as={[Function]}
          control={[Function]}
          label="Dosis"
          name="dose"
          onChange={[Function]}
          placeholder="Dosis"
          value=""
          width={2}
        />
        <FormInput
          as={[Function]}
          control={[Function]}
          label="Medicamento"
          name="medicine"
          onChange={[Function]}
          placeholder="Medicamento"
          value=""
          width={6}
        />
        <FormInput
          as={[Function]}
          control={[Function]}
          label="Frecuencia"
          name="frequency"
          onChange={[Function]}
          placeholder="Frecuencia"
          value=""
          width={4}
        />
        <FormInput
          as={[Function]}
          control={[Function]}
          label="Duración"
          name="duration"
          onChange={[Function]}
          placeholder="Duración"
          value=""
          width={3}
        />
        <Treatment__AddIconContainer>
          <Treatment__AddIcon
            name="plus"
            onClick={[Function]}
          />
        </Treatment__AddIconContainer>
      </FormGroup>
      <Treatment__ErrorMessage
        name="error"
      />
      <Table
        as="table"
        basic="very"
      >
        <TableBody
          as="tbody"
        />
      </Table>
      <Treatment__FormTextArea
        label="Indicaciones del tratamiento:"
        name="indications"
        placeholder="Escribe aquí las indicaciones del tratamiento."
      />
    </Form>
  </CustomCard>
</Fragment>
`;
